src/lib/FreeC/Backend/Agda/Pretty.hs

```diff
@@ -0,0 +1,44 @@
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+-- | This module contains 'Pretty' instances for the Agda AST.
+--
+--   The Agda has its own 'Agda.Pretty' class in 'Agda.Utils.Pretty'. Therefore
+--   we cannot instance every @a@ with an @Agda.Pretty@ instance with our
+--   'Pretty' instance without generating overlapping instances for basic types
+--   (e.g. 'String').
+--   We introduce the wrapper datatype 'PrettyAgda' to avoid the overlapping
+--   instances and explicitly declare 'Pretty' instances for for common AST
+--   nodes.
+module FreeC.Backend.Agda.Pretty
+  ( prettyAgda
+  )
+where
+
+-- import qualified Agda.Syntax.Concrete.Generic  as Agda
```

Comment:
When commenting out code, try to use block comments to separate them from regular comments. Is there a reason why you kept this import declaration?

```suggestion
{- import qualified Agda.Syntax.Concrete.Generic  as Agda -}
```